flink-core/src/main/java/org/apache/flink/configuration/HistoryServerOptions.java

```diff
@@ -143,5 +143,20 @@ public class HistoryServerOptions {
                                             code("IllegalConfigurationException"))
                                     .build());
 
+    public static final ConfigOption<String> HISTORY_SERVER_STORAGE_BACKEND =
```

Comment:
I suggest adding something to the documentation around this new option in [https://nightlies.apache.org/flink/flink-docs-release-1.20/docs/deployment/advanced/historyserver/](https://nightlies.apache.org/flink/flink-docs-release-1.20/docs/deployment/advanced/historyserver/). The docs could include details on how to extend ArchiveFetcher - as per the javadoc at the top of that class.